nvflare/private/fed/server/fed_server.py

```diff
@@ -509,18 +509,19 @@ def client_heartbeat(self, request: Message) -> Message:
             if self.admin_server:
                 self.admin_server.client_heartbeat(token, client_name)
 
-            abort_runs = self._sync_client_jobs(request, token)
             reply = self._generate_reply(
                 headers={CellMessageHeaderKeys.MESSAGE: "Heartbeat response"}, payload=None, fl_ctx=fl_ctx
             )
-            if abort_runs:
-                reply.set_header(CellMessageHeaderKeys.ABORT_JOBS, abort_runs)
-
-                display_runs = ",".join(abort_runs)
-                self.logger.debug(
-                    f"These jobs: {display_runs} are not running on the server. "
-                    f"Ask client: {client_name} to abort these runs."
-                )
+            if isinstance(self.server_state, HotState):
```

Comment:
No need to make the change for this HotState check. The "self._handle_state_check(state_check, fl_ctx)" before already ensure only HotState can reach here.

Reply:
"handle_state_check" only sets a prop inside fl_ctx.

Cold state can also reach here.

Reply:
You are right. But the proper fix is to do something like this:

    def _handle_state_check(self, state_check, fl_ctx: FLContext):
        if state_check.get(ACTION) in [NIS, ABORT_RUN]:
            fl_ctx.set_prop(FLContextKey.COMMUNICATION_ERROR, state_check.get(MESSAGE), sticky=False)
            return False
        return True

And:

            if self._handle_state_check(state_check, fl_ctx):
                return